crates/remora/src/executor/fake.rs

```diff
@@ -0,0 +1,196 @@
+
+use std::{collections::BTreeMap, time::{Duration, Instant}};
+use rand::thread_rng;
+use rand_distr::{Distribution, Normal};
+
+use sui_types::base_types::VersionNumber;
+use super::api::{ExecutableTransaction, ExecutionResults, Executor, StateStore, Transaction};
+use sui_single_node_benchmark::benchmark_context::BenchmarkContext;
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SequenceNumber}, committee::EpochId, digests::TransactionDigest, effects::TransactionEffects, error::SuiResult, object::{Object, Owner}, storage::{BackingPackageStore, ChildObjectResolver, ObjectStore, PackageObject, ParentSync}, transaction::InputObjectKind
+};
+
+
+#[derive(Clone)]
+pub struct FakeTransaction {
+    delay: u64, // delay in ms for that this transaction simulates
+}
+pub type FakeTransactionType = Transaction<FakeExecutor>;
+
+pub type FakeExecutionResults = ExecutionResults<FakeExecutor>;
+pub struct FakeStore;
+
+impl FakeExecutionResults {
+    // todo: is there a way to make this a constant instead of a fn?
+    pub fn dummy() -> Self {
+        FakeExecutionResults {
+            updates: TransactionEffects::default(),
+            new_state: BTreeMap::new(),
+        }
+    }
+}
+
+impl ExecutableTransaction for FakeTransaction {
+    fn digest(&self) -> &TransactionDigest {
+        // Return a reference to a static TransactionDigest
+        &TransactionDigest::ZERO
+    }
+
+    fn input_objects(&self) -> Vec<InputObjectKind> {
```

Comment:
Perhaps for a next PR (you did good to now bring all the complexity at once), but we would need to vary the number of inputs for this transaction (so we can test our resharding strategy)